### PR TITLE
Add response compression for static assets to the dashboard

### LINKED ISSUE #4150: Dashboard should configure response compression & caching for static assets
The dashboard currently doesn't configure compression or caching headers for the static assets it delivers, including libraries like plotly (which is [3.5 MB](https://github.com/dotnet/aspire/issues/4149)).

We should configure the response compression middleware to ensure that at least static assets are compressed (and potentially the HTML output too), and investigate setting cache headers for static assets with versioned paths to help improve the client-side experience.

### PR DESCRIPTION
**Pull request by DamianEdwards**
Fixes #4150
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/4153)

### CODE REVIEW

**Review by drewnoakes — APPROVED** [review]

**Comment by DamianEdwards** [comment]
@davidfowl @blowdart any comments on this one? It's pretty straightforward but there's nuance in where in the middleware pipeline it's enabled and enabling compression over HTTPS but not for HTML, to mitigate any security issues stemming from being able to force user content to be compressed over HTTPS.

**Comment by blowdart** [comment]
Funnily enough Javier is doing the same thing as a feature - you should sync with him for his new idea which might mitigate the need to do this in a custom manner. (But generally, no, no concerns for this type of thing as it's static assets)

File: src/Aspire.Dashboard/DashboardWebApplication.cs [hunk 1]
```diff
@@ -121,6 +121,12 @@ public DashboardWebApplication(Action<WebApplicationBuilder>? configureBuilder =
         // Add services to the container.
         builder.Services.AddRazorComponents().AddInteractiveServerComponents();
         builder.Services.AddCascadingAuthenticationState();
+        builder.Services.AddResponseCompression(options =>
+        {
+            options.EnableForHttps = true;
+            // Avoid compressing HTML to mitigate user supplied data being compressed over HTTPS
```
Comment (danmoseley) [inline src/Aspire.Dashboard/DashboardWebApplication.cs:127]: Should the comment say because of side channel attack? Or maybe that's obvious.

Comment (DamianEdwards) [reply src/Aspire.Dashboard/DashboardWebApplication.cs:127]: Maybe just link to [our docs about it](https://learn.microsoft.com/aspnet/core/performance/response-compression?view=aspnetcore-8.0#compression-with-https)?

Comment (danmoseley) [reply src/Aspire.Dashboard/DashboardWebApplication.cs:127]: Seems reasonable

Comment (JamesNK) [reply src/Aspire.Dashboard/DashboardWebApplication.cs:127]: What are the rules when it comes to Blazor compression? Are they different from regular server rendered HTML?

We only want static resources compressed:
* JavaScript
* CSS
* Images
* SVG

Everything else should be excluded:
* Communication over SignalR websockets
* gRPC calls (remember the dashboard has some gRPC endpoints)
* Re-rendered HTML (I think that's disabled)

Comment (JamesNK) [reply src/Aspire.Dashboard/DashboardWebApplication.cs:127]: I'd feel safer with an allow list than a deny list.

Comment (DamianEdwards) [reply src/Aspire.Dashboard/DashboardWebApplication.cs]: @JamesNK changed it to use an allow list

**Review by JamesNK — APPROVED** [review]

**Pull request merged**